Final

tutur90 opened the pull request:
If you need that i improve something, feel free to email me

Commit:
fix_bug

ZhangRui111 commented:
Hi,

I’m happy to see a new pull request from you. Well done! I have already merged the pull request you submitted earlier. However, for the sake of maintaining code consistency and other reasons, I made a few modifications. For example, 

- I deleted the `core/utils.py` file, and the related functions were moved directly to `demo9.py` and `demo10.py`. I believe that maintaining the `core/` directory as lean as possible, keeping only the most essential functionality, is a better approach. The functions in `core/utils.py` are not core to the simulation platform; they seem more like user-written code when using the platform.

- I merged `demo8.py` and `demo9.py`, which led to `demo10.py` being renamed to `demo9.py` and `demo11.py` being renamed to `demo10.py`. Please keep this in mind when submitting a new pull request.

I also have a few confusions. Maybe you can help me clear up my confusion.

- I kept the functionality code for the **refresh rate** but hid its calling interface. I’m unclear about the motivation and effect of the refresh rate. In some `demoX.py` files, setting the refresh rate to < 1 seems to slow down the simulation without affecting the log information or results. Conversely, setting the refresh rate > 1 directly causes some `demoX.py` files to fail.

- When using the Pakistan dataset (`demo9.py`, `demo10.py`), why is the code for starting a new task `if env.now >= generated_time:` instead of `if env.now == generated_time:`?

- The current DQRL (`demo10.py`) generates the following warning message during execution:
  ```
  UserWarning: Using a target size (torch.Size([1])) that is different from the input size (torch.Size([])). This will likely lead to incorrect results due to broadcasting. Please ensure they have the same size.
  ```
  Additionally, the training results don’t seem to be performing well. If you have time, I’d appreciate it if you could debug the strategy and example code further. If the training time with the Pakistan dataset is too long, you might want to try using Topo4MEC, since this is just demo code.

If the modifications I made to the earlier pull request caused any errors, please feel free to let me know.

---
Regarding the new pull request, I have one suggestion: please keep the empty `__init__.py` file. 

Based on the above changes, the new pull request (#22) conflicts quite a bit with the current code. I suggest syncing the code first and then submitting a new pull request. If you have any alternative methods, please let me know.

Finally, thank you again for your contributions. **I wish you all the best!**

tutur90 commented:
Hi,

Sorry for the late reply — I’ve been really busy with university lately.

To answer your question:

   -

   The key idea behind the refresh rate is to keep it in the same order of
   magnitude as the generation time. This works well for my dataset
   (generation time around 0.01s), and it can also help improve precision in
   other examples. For my dataset, it significantly changes the results.
   -

   env.now >= generated_time is necessary when the refresh rate is greater
   than 1. Since env.now is a float, using == may not always hold due to
   decimal precision issues, which could lead to an infinite loop.
   -

   The user warning has been fixed.
   -

   The DQRL algorithm has been corrected and now works on the topo4mec. The
   results aren’t very impressive yet, but it does run.
   -

   I’ve included the execution energy in the Task class so that the energy
   consumed by each task can be calculated. According to Leila Ismail and
   Huned Materwala in *Computing Server Power Modeling in a Data Center:
   Survey, Taxonomy, and Performance Evaluation*, ACM Computing Surveys,
   53(3):1–34, May 2021, execution energy is most likely linearly proportional
   to CPU consumption rather than proportional to the cube. But if you prefer
   to keep it as is, you can just uncomment the relevant line in the Task
   class.

Let me know if there is any problem with the last version.

Thank you for your framework — it was really helpful for one of my
projects, and I appreciate the opportunity to contribute. Wishing you all
the best!

Best regards,
Arthur

Le lun. 3 mars 2025 à 15:50, ZhangRui ***@***.***> a écrit :

> Hi,
>
> I’m happy to see a new pull request from you. Well done! I have already
> merged the pull request you submitted earlier. However, for the sake of
> maintaining code consistency and other reasons, I made a few modifications.
> For example,
>
>    -
>
>    I deleted the core/utils.py file, and the related functions were moved
>    directly to demo9.py and demo10.py. I believe that maintaining the
>    core/ directory as lean as possible, keeping only the most essential
>    functionality, is a better approach. The functions in core/utils.py
>    are not core to the simulation platform; they seem more like user-written
>    code when using the platform.
>    -
>
>    I merged demo8.py and demo9.py, which led to demo10.py being renamed
>    to demo9.py and demo11.py being renamed to demo10.py. Please keep this
>    in mind when submitting a new pull request.
>
> I also have a few confusions. Maybe you can help me clear up my confusion.
>
>    -
>
>    I kept the functionality code for the *refresh rate* but hid its
>    calling interface. I’m unclear about the motivation and effect of the
>    refresh rate. In some demoX.py files, setting the refresh rate to < 1
>    seems to slow down the simulation without affecting the log information or
>    results. Conversely, setting the refresh rate > 1 directly causes some
>    demoX.py files to fail.
>    -
>
>    When using the Pakistan dataset (demo9.py, demo10.py), why is the code
>    for starting a new task if env.now >= generated_time: instead of if
>    env.now == generated_time:?
>    -
>
>    The current DQRL (demo10.py) generates the following warning message
>    during execution:
>
>    UserWarning: Using a target size (torch.Size([1])) that is different from the input size (torch.Size([])). This will likely lead to incorrect results due to broadcasting. Please ensure they have the same size.
>
>    Additionally, the training results don’t seem to be performing well.
>    If you have time, I’d appreciate it if you could debug the strategy and
>    example code further. If the training time with the Pakistan dataset is too
>    long, you might want to try using Topo4MEC, since this is just demo code.
>
> If the modifications I made to the earlier pull request caused any errors,
> please feel free to let me know.
> ------------------------------
>
> Regarding the new pull request, I have one suggestion: please keep the
> empty __init__.py file.
>
> Based on the above changes, the new pull request (#22
> <https://github.com/ZhangRui111/RayCloudSim/pull/22>) conflicts quite a
> bit with the current code. I suggest syncing the code first and then
> submitting a new pull request. If you have any alternative methods, please
> let me know.
>
> Finally, thank you again for your contributions. *I wish you all the
> best!*
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ZhangRui111/RayCloudSim/pull/22#issuecomment-2694655463>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A5SQAISF4H5BDYCVSS5ZHD32SRT3ZAVCNFSM6AAAAABYFN23GSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMOJUGY2TKNBWGM>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
> [image: ZhangRui111]*ZhangRui111* left a comment
> (ZhangRui111/RayCloudSim#22)
> <https://github.com/ZhangRui111/RayCloudSim/pull/22#issuecomment-2694655463>
>
> Hi,
>
> I’m happy to see a new pull request from you. Well done! I have already
> merged the pull request you submitted earlier. However, for the sake of
> maintaining code consistency and other reasons, I made a few modifications.
> For example,
>
>    -
>
>    I deleted the core/utils.py file, and the related functions were moved
>    directly to demo9.py and demo10.py. I believe that maintaining the
>    core/ directory as lean as possible, keeping only the most essential
>    functionality, is a better approach. The functions in core/utils.py
>    are not core to the simulation platform; they seem more like user-written
>    code when using the platform.
>    -
>
>    I merged demo8.py and demo9.py, which led to demo10.py being renamed
>    to demo9.py and demo11.py being renamed to demo10.py. Please keep this
>    in mind when submitting a new pull request.
>
> I also have a few confusions. Maybe you can help me clear up my confusion.
>
>    -
>
>    I kept the functionality code for the *refresh rate* but hid its
>    calling interface. I’m unclear about the motivation and effect of the
>    refresh rate. In some demoX.py files, setting the refresh rate to < 1
>    seems to slow down the simulation without affecting the log information or
>    results. Conversely, setting the refresh rate > 1 directly causes some
>    demoX.py files to fail.
>    -
>
>    When using the Pakistan dataset (demo9.py, demo10.py), why is the code
>    for starting a new task if env.now >= generated_time: instead of if
>    env.now == generated_time:?
>    -
>
>    The current DQRL (demo10.py) generates the following warning message
>    during execution:
>
>    UserWarning: Using a target size (torch.Size([1])) that is different from the input size (torch.Size([])). This will likely lead to incorrect results due to broadcasting. Please ensure they have the same size.
>
>    Additionally, the training results don’t seem to be performing well.
>    If you have time, I’d appreciate it if you could debug the strategy and
>    example code further. If the training time with the Pakistan dataset is too
>    long, you might want to try using Topo4MEC, since this is just demo code.
>
> If the modifications I made to the earlier pull request caused any errors,
> please feel free to let me know.
> ------------------------------
>
> Regarding the new pull request, I have one suggestion: please keep the
> empty __init__.py file.
>
> Based on the above changes, the new pull request (#22
> <https://github.com/ZhangRui111/RayCloudSim/pull/22>) conflicts quite a
> bit with the current code. I suggest syncing the code first and then
> submitting a new pull request. If you have any alternative methods, please
> let me know.
>
> Finally, thank you again for your contributions. *I wish you all the
> best!*
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ZhangRui111/RayCloudSim/pull/22#issuecomment-2694655463>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A5SQAISF4H5BDYCVSS5ZHD32SRT3ZAVCNFSM6AAAAABYFN23GSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMOJUGY2TKNBWGM>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
